### PR TITLE
build: allign mypy cfg and ver to core

### PR DESCRIPTION
**Pull request by chemelli74**

### CODE REVIEW

File: custom_components/midea_ac_lan/config_flow.py [hunk 1]
```diff
@@ -196,7 +200,8 @@ async def async_step_login(
                 )
                 if self.cloud is None:
                     raise AbortFlow(
-                        f"Can not get midea cloud: {SERVERS[user_input[CONF_SERVER]]}"
+                        f"Can not get midea cloud: {
```
Comment (Necroneco) [inline custom_components/midea_ac_lan/config_flow.py:203]: Python version 3.11 does not allow new lines in expression parts of non-triple-quoted f-strings

**Review by Necroneco — CHANGES_REQUESTED** [review]

File: custom_components/midea_ac_lan/config_flow.py (hunk 1 above)
Comment (Necroneco) [reply custom_components/midea_ac_lan/config_flow.py:203]: Do this in later PR

**Review by Necroneco — APPROVED** [review]

File: custom_components/midea_ac_lan/config_flow.py [hunk 2]
```diff
@@ -275,7 +277,7 @@ async def async_step_discovery(
                     self.available_device[
                         device_id
                     ] = f"{device_id} ({self.supports.get(
-                        device.get(CONF_TYPE))})"
+                            device.get(CONF_TYPE))})"
```
Comment (Necroneco) [inline custom_components/midea_ac_lan/config_flow.py:280]: another need fmt: off

**Review by Necroneco — DISMISSED** [review]

File: custom_components/midea_ac_lan/config_flow.py (hunk 2 above)
Comment (chemelli74) [reply custom_components/midea_ac_lan/config_flow.py]: Done

**Review by Necroneco — APPROVED** [review]

**Pull request merged**